src/storage/v2/storage_error.hpp

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#pragma once
+
+#include "storage/v2/constraints.hpp"
+
+#include <variant>
+
+namespace memgraph::storage {
+
+enum class ReplicationError {
+  UNABLE_TO_SYNC_REPLICATE,
+};
+
+struct StorageDataManipulationError {
+  std::variant<ConstraintViolation, ReplicationError> error;
+
+  bool operator==(const StorageDataManipulationError &) const = default;
+};
+
+enum class IndexDefinitionError {
+  EXISTANT_INDEX,
+  NONEXISTANT_INDEX,
+};
```

Comment:
I have checked and you are 100% that it doesn't bring much value since in any case we get an error we know what we are getting.
My reasoning to keep it would be: as we now return different error (and assert on it in the interpreter), I would be very happy to get this assert raised if I ever incorrectly modify the code beneath index.hpp.
But while I was checking the code I see that this IndexDefinitionError does not bring anything else but this assert.

As conclusion I think we can safely remove it and I will see if the error structure can't even be simplified further.

Reply:
Note: I have done the same for `ConstraintDefinitionError`